pkg/debug/spec.sh

```diff
@@ -1,11 +1,73 @@
 #!/bin/sh
+# This script creates an initial hardware model file
+# The output is conservative when it comes to assignment groups in that
+# it does not form any assignment groups but merely checks whether there are
+# multiple functions on the same controller, and in that case the assignment
+# group is set to empty.
 
 if [ "$(uname -m)" = x86_64 ]; then
    ARCH=2
 else
    ARCH=4
 fi
 
+# log_to_dev returns the bus:controller without the function value
+# $1 is the PciLong value
+long_to_dev() {
+    #shellcheck disable=SC2039
+    local pcilong=$1
+    #shellcheck disable=SC2039
+    local dom=${pcilong%%:*}
+    #shellcheck disable=SC2039
+    local rest=${pcilong#"$dom":}
+    #shellcheck disable=SC2039
+    local bus=${rest%%:*}
+    rest=${rest#"$bus":}
+    #shellcheck disable=SC2039
+    local dev=${rest%%:*}
+    #shellcheck disable=SC2039
+    local ct=${dev%%.*}
+    # local fn=${dev##*.}
+    echo "$bus:$ct"
+}
+
+# check_multifunction returns 0 if the controller has multiple functions
+# $1 is the PciLong value
+check_multifunction() {
+    #shellcheck disable=SC2039
+    local pcilong=$1
+    #shellcheck disable=SC2039
+    local short_no_fn
+    short_no_fn=$(long_to_dev "$pcilong")
+    count=$(lspci | grep -c ^"$short_no_fn")
+    if [ "$count" = 0 ]; then
```

Comment:
Just curious why did you feel that this part of if is needed?

Reply:
I started with just checking that the function number is zero. But we can have boxes where e.g., the audio controler is X:Y:Z.0 and some memory controller is X:y:Z.1. Could also be the case for other integrated devices like Ethernets in some common northbridge. Don't think I've seem such cases on real hardware (I've seen the case of X:Y:Z.1 etc for the device we care about reporting though).